apps/core/src/hooks/useGetObject.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useRpcClient } from '../api/RpcClientContext';
+import { normalizeSuiAddress, type SuiObjectDataOptions } from '@mysten/sui.js';
+import { useQuery } from '@tanstack/react-query';
+
+const defaultOptions: SuiObjectDataOptions = {
+    showType: true,
+    showContent: true,
+    showOwner: true,
+    showDisplay: true,
+};
+
+export function useGetObject(
+    objectId?: string | null,
+    options = defaultOptions
+) {
+    const rpc = useRpcClient();
+    const normalizedObjId = objectId && normalizeSuiAddress(objectId);
+    return useQuery(
+        ['object', normalizedObjId],
+        async () =>
```

Comment:
does it need to be declared as async if you're just returning the promise here?

Reply:
Good catch..